server/cluster/coordinator.go

```diff
@@ -947,7 +946,14 @@ func (s *scheduleController) AllowSchedule(diagnosable bool) bool {
 		}
 		return false
 	}
-	if s.IsPaused() || s.cluster.GetUnsafeRecoveryController().IsRunning() {
+	allowed, _ := s.cluster.checkSchedulingAllowance()
+	if !allowed {
+		if diagnosable {
+			s.diagnosticRecorder.setResultFromStatus(halted)
```

Comment:
Please help take a look here, thx. @CabinfeverB

Reply:
This part LGTM